### ISSUE TITLE
Commas do not compile in arbitrary values with postcss 7 compat

### ISSUE THREAD

**Issue by Soviut** [tailwindlabs/tailwindcss#6119]
**What version of Tailwind CSS are you using?**

postcss compat 2.2.17

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss-cli 7.0.39, webpack 2.7.0

**What version of Node.js are you using?**

v14.0.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Ubuntu 20 under WSL2

**Reproduction URL**

Unfortunately this works in Tailwind Play v2.2.15 so I can't reproduce it online.

**Describe your issue**

I'm trying to transfer a design I mocked up in Tailwind Play (v2.2.15) to a project running Tailwind 2.2.17 postcss 7 compat. The code includes an arbitrary value to set grid row templates.

```html
<div class="grid grid-rows[repeat(2,min-content)]">
```

In Tailwind Play and a clean project running postcss 8 under Vite, the code compiles and produces the correct style.

![image](https://user-images.githubusercontent.com/226882/142345422-326d5ae3-396e-4fe8-9c1f-1604856d46bc.png)

However, in postcss 7 compat, the style is ignored seemingly because of the comma. When I change the arbitrary value to the following:

```html
<div class="grid grid-rows-[min-content]">
```

It compiles in postcss 7 as expected

![image](https://user-images.githubusercontent.com/226882/142346050-f56fde0e-6d95-4215-aae5-380303fdaaec.png)

And another example with lengths and a comma does not compile either

```html
<div class="grid grid-rows-[200px,400px]">
```

My `tailwind.config.js`  on the postcss 7 compat project is as follows.

```javascript
module.exports = {
  mode: 'jit',
  purge: ['./index.html', './src/**/*.{vue,js,ts,jsx,tsx}'],
  corePlugins: {
    // TODO: disable modern-normalize reset to play nice with old reset
    preflight: false,
  },
  darkMode: false, // or 'media' or 'class'
  important: true,
  // ...
}
```

**Comment by harvanchik** [comment]
Can confirm that I am also now getting this issue on Tailwind V2.2.19 and PostCSS V8.3.11.

**Comment by adamwathan** [comment]
Hey! Unfortunately going forward we will no longer be providing or supporting the PostCSS 7 compatibility build as it's a real headache to maintain and CRA is pretty much the only reason anyone is ever using PostCSS 7 now.

I would recommend migrating to the Tailwind CLI tool which uses PostCSS 8 under the hood, and using a tool like `npm-run-all` or `concurrently` to just build Tailwind in parallel with your existing build setup so you aren't stuck with PostCSS 7. Hopefully whatever underlying tool you are using updates to PostCSS 8 soon (it's well over a year old at this point) and then you can just use the regular version of Tailwind.

Closing as a "won't fix" — sorry 😔

**Comment by Soviut** [comment]
@adamwathan Sorry to hear that. While I agree that a cutoff is warranted with tailwind 3 on the horizon, it seems strange that a 2.2.17 build of the postcss 7 compat exists with the issue when the postcss 8 version does not. Why not simply set the version cutoff sooner on the compat releases?

Sadly, this is a legacy Vue 2 project built on a custom Webpack 2.7 config and no Vue CLI. It's dependency hell and upgrading one dependency often cascades into a compatibility log jam.

I'll give the tailwind CLI "sidecar" idea a shot and report back here if I works.
